builtin/credential/approle/path_role.go

```diff
@@ -718,6 +722,27 @@ func (b *backend) roleEntry(ctx context.Context, s logical.Storage, roleName str
 		return nil, err
 	}
 
+	needsUpgrade := false
+
+	if role.BoundCIDRListOld != "" {
+		role.BoundCIDRList = strings.Split(role.BoundCIDRListOld, ",")
+		role.BoundCIDRListOld = ""
+		needsUpgrade = true
+	}
+
+	if needsUpgrade && (b.System().LocalMount() || !b.System().ReplicationState().HasState(consts.ReplicationPerformanceSecondary)) {
```

Comment:
Not sure if the second part of the check (i.e. `(b.System().LocalMount() || !b.System().ReplicationState().HasState(consts.ReplicationPerformanceSecondary))`) is required since `roleEntry` should be called only from a primary after the request gets forwarded and routed.

Reply:
If the secondary is doing what would otherwise only be a read-only operation and this check isn't there and it is not in contact with the primary, the call will fail until of succeed.